src/main/java/teammates/ui/controller/Action.java

```diff
@@ -161,10 +161,10 @@ private void validateOriginIfRequired() {
 
         String referrer = request.getHeader("referer");
         if (referrer == null) {
-            throw new InvalidOriginException("Missing HTTP referrer");
-        }
-
-        if (!isHttpReferrerValid(referrer)) {
+            // Requests with missing referrer information are given the benefit of the doubt to
+            // accommodate users who choose to disable the HTTP referrer setting in their browser
+            // for privacy reasons
```

Comment:
Probably one of the few times where an effectively empty if block makes sense 😅

Reply:
For the purpose of documentation haha. Checkstyle only allows an empty `if` block if it contains comments.